Added support for viewing markdown documents.

sarmadka opened the pull request:
* Added MarkdownTranslator utility class for converting from markdown to html.
* Added DocView widget for rendering full documents from html or markdown sources.
* Added RegExp class and renamed Label to Text.
* Fixed some bugs with regular expressions.
* Allow class based styling on the component itself rather than only on its children.
* Enabled differentiating between immediate children and non immediate children when defining styles.
* Enabled styling children by tag name.
* Fixed a bug with RoutingSwitcher causing unnecessary instantiation of components when route events result in staying on the same component.
* Added some missing Arabic localizations and moved utility files into Utils folder.
* Updated the routing example to use DocView for displaying MD formatted text.

sarmadka committed:
Added RegExp class and renamed Label to Text.

Also added some missing Arabic localizations and moved utility
files into Utils folder.

sarmadka committed:
Added more control over styling.

* Allow class based styling on the component itself rather
  than only on its children.
* Enabled differentiating between immediate children and
  non immediate children when defining styles.
* Enabled styling children by tag name.

sarmadka committed:
Fixed a bug with regular expressions.

sarmadka committed:
Fixed a bug with RoutingSwitcher.

Fixed a bug with RoutingSwitcher causing unnecessary
instantiation of components when route events result in
staying on the same component.

sarmadka committed:
Added MarkdownTranslator and DocView.

* Added MarkdownTranslator utility class for converting from
  markdown to html.
* Added DocView widget for rendering full documents from html
  or markdown sources.
* Fixed an issue with regular expressions when matching
  multibyte characters.
* Updated the routing example to use DocView for displaying
  MD formatted text.

sarmadka committed:
Fixed an issue with MarkdownTranslator.

SlemanAlrstum231 commented on WebPlatform/Utils/string_operations.alusus:
```diff
@@ -0,0 +1,30 @@
+
+    function matchRegex (str: ptr[array[Char]], regexId: ArchInt): Array[String] {
+        def lastIndex: Int;
+        return matchRegex(str, regexId, lastIndex);
```
سؤال اخي الكريم هنا يمرر بشكل افتراضي int بالعنوان

sarmadka replied:
لم أفهم السؤال، ماذا تقصد بالعنوان؟

SlemanAlrstum231 commented on WebPlatform/Utils/string_operations.alusus:
```diff
@@ -0,0 +1,30 @@
+    }
+
+    function matchRegex (str: ptr[array[Char]], regexId: ArchInt, lastIndex: ref[Int]): Array[String] {
+        return parseRegexMatchResult(_matchRegex(str, 0, regexId, lastIndex));
```
واني تعرفي هذا التابع اخي الكريم _matchRegex

sarmadka replied:
تعريفه في ملف browser_api

sarmadka committed:
Improved the performance of MarkdownTranslator.

sarmadka committed:
Fixed another issue with MarkdownTranslator.

sarmadka committed:
Added support for `float` style.

sarmadka committed:
Fixed a couple of issues with Box and Image.

sarmadka committed:
Added new properties and functions.

* Added Window.hash for getting the hash value from address bar.
* Added Window.query for getting the query from the address bar.
* Added Widget.scrollIntoView.

sarmadka committed:
Fixed an issue with regular expressions.

sarmadka committed:
Fixed an issue with Component id.

Component ID was being set to the ID of the parent widget
rather than the ID of its view.

sarmadka committed:
Fixed some Arabic aliases.

sarmadka committed:
Fixed an issue with Component rebuild.

Fixed an issue with components where updating the view does not
result in rebuilding the component.

sarmadka committed:
Made RoutingStack and RoutingSwitcher work with widgets.

Previously RoutingStack and RoutingSwitcher would only accept
components. This change makes it also possible to pass simple
widgets.

sarmadka committed:
Added fontWeight to Style.

sarmadka committed:
Update for compatibility with Http@0.1.0.

sarmadka committed:
Replace # with $ for styling HTML elements.

This is to avoid ambiguity with CSS's usage of #, so that it is
clear to the user that this symbol is not for selecting by ID.
